test/e2e/network/netpol/probe.go

```diff
@@ -83,11 +89,24 @@ func ProbePodToPodConnectivity(k8s *kubeManager, model *Model, testCase *TestCas
 
 // probeWorker continues polling a pod connectivity status, until the incoming "jobs" channel is closed, and writes results back out to the "results" channel.
 // it only writes pass/fail status to a channel and has no failure side effects, this is by design since we do not want to fail inside a goroutine.
-func probeWorker(k8s *kubeManager, jobs <-chan *ProbeJob, results chan<- *ProbeJobResults, timeoutSeconds int) {
+func probeWorker(prober Prober, jobs <-chan *ProbeJob, results chan<- *ProbeJobResults, timeoutSeconds int) {
 	defer ginkgo.GinkgoRecover()
 	for job := range jobs {
 		podFrom := job.PodFrom
-		connected, command, err := k8s.probeConnectivity(podFrom.Namespace, podFrom.Name, podFrom.Containers[0].Name(), job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain), job.Protocol, job.ToPort, timeoutSeconds)
+		if job.PodTo.ServiceIP == "" {
+			results <- &ProbeJobResults{
+				Job:         job,
+				IsConnected: false,
+				Err:         fmt.Errorf("empty service ip"),
+			}
+		}
+		// note that we can probe a dnsName instead of ServiceIP by using dnsName like so:
+		// we stopped doing this because we wanted to support netpol testing in non dns enabled
+		// clusters, but might re-enable it later.
+		// dnsName := job.PodTo.QualifiedServiceAddress(job.ToPodDNSDomain)
+
+		// TODO how does ipv6 work here?
```

Comment:
For single-stack IPv6 you don't need to do anything.

In dual-stack clusters, ideally all tests would check both IPv4 and IPv6 connectivity. (ie, that you can connect to both `ClusterIPs` values when NetworkPolicy is supposed to allow the connection, and that you can't connect to either `ClusterIPs` value when NetworkPolicy is supposed to block the connection

Reply:
fixed to reflect this